File: fortran/src/arrays.f90
module arrays_mod
    implicit none

contains
    subroutine demonstrate_arrays()
        ! Array declarations
        integer, dimension(5) :: numbers = [1, 2, 3, 4, 5]
        real, dimension(3,3) :: matrix
        character(len=10), dimension(4) :: fruits
        integer :: i, j
        
        print *, '=== Arrays ==='
        
        ! Initialize arrays
        fruits = ['apple    ', 'banana   ', 'orange   ', 'grape    ']
        matrix = reshape([1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0], [3,3])
        
        ! Array operations
        print *, 'Original numbers:', numbers
        print *, 'Numbers * 2:', numbers * 2
        print *, 'Sum of numbers:', sum(numbers)
        print *, 'Maximum value:', maxval(numbers)
        print *, 'Minimum value:', minval(numbers)
        
        ! Array slicing
        print *, 'First three numbers:', numbers(1:3)
        print *, 'Every second number:', numbers(1:5:2)
        
        ! 2D array operations
        print *, 'Matrix:'
        do i = 1, 3
            print *, (matrix(i,j), j=1,3)
        end do
        
        ! Array of strings
        print *, 'Fruits:'
        do i = 1, size(fruits)
            print *, trim(fruits(i))
        end do
        
        ! Array functions
        print *, 'Array size:', size(numbers)
        print *, 'Matrix shape:', shape(matrix)
        
        ! Array allocation (dynamic arrays)
        block
            integer, allocatable :: dynamic_array(:)
            allocate(dynamic_array(3))
            dynamic_array = [10, 20, 30]
            print *, 'Dynamic array:', dynamic_array
            deallocate(dynamic_array)
        end block
    end subroutine demonstrate_arrays
end module arrays_mod

File: ruby/lib/arrays.rb
module Arrays
  def self.demonstrate
    puts "\n=== Arrays ==="
    
    # Creating arrays
    numbers = [10, 20, 30, 40]
    fruits = %w[apple banana orange grape]  # %w creates array of strings
    mixed = [1, "two", 3.0, :four]         # Arrays can hold different types
    
    puts "Basic array operations:"
    puts "Numbers: #{numbers}"
    puts "Fruits: #{fruits}"
    puts "Mixed: #{mixed}"
    
    # Accessing elements
    puts "\nAccessing elements:"
    puts "First fruit: #{fruits.first}"
    puts "Last fruit: #{fruits.last}"
    puts "Second number: #{numbers[1]}"
    puts "Last number (negative index): #{numbers[-1]}"
    
    # Array slicing
    puts "\nArray slicing:"
    puts "First two fruits: #{fruits[0,2]}"
    puts "Last two fruits: #{fruits[-2..-1]}"
    
    # Array methods
    puts "\nArray methods:"
    puts "Array length: #{fruits.length}"
    puts "Array includes 'banana'? #{fruits.include?('banana')}"
    puts "Array reversed: #{fruits.reverse}"
    puts "Original array (not modified): #{fruits}"
    
    # Modifying arrays
    puts "\nModifying arrays:"
    fruits.push("mango")            # Add to end
    puts "After push: #{fruits}"
    fruits.unshift("pineapple")     # Add to beginning
    puts "After unshift: #{fruits}"
    popped = fruits.pop             # Remove from end
    puts "Popped: #{popped}"
    shifted = fruits.shift          # Remove from beginning
    puts "Shifted: #{shifted}"
    puts "Final array: #{fruits}"
    
    # Array operations
    puts "\nArray operations:"
    nums1 = [1, 2, 3]
    nums2 = [4, 5, 6]
    puts "Concatenation: #{nums1 + nums2}"
    puts "Multiplication: #{[1, 2] * 3}"
    
    # Iterating arrays
    puts "\nIterating arrays:"
    fruits.each_with_index do |fruit, index|
      puts "#{index + 1}. #{fruit}"
    end
    
    # Array transformations
    puts "\nArray transformations:"
    squares = numbers.map { |n| n ** 2 }
    puts "Original numbers: #{numbers}"
    puts "Squared numbers: #{squares}"
    
    # Array filtering
    evens = numbers.select { |n| n.even? }
    puts "\nEven numbers: #{evens}"
    
    # Array sorting
    puts "\nSorting:"
    puts "Sorted fruits: #{fruits.sort}"
    puts "Reverse sorted fruits: #{fruits.sort.reverse}"
  end
end